pcsx2/DEV9/AdapterUtils.h

```diff
@@ -47,4 +47,5 @@ namespace AdapterUtils
 	std::vector<PacketReader::IP::IP_Address> GetGateways(ifaddrs* adapter);
 	std::vector<PacketReader::IP::IP_Address> GetDNS(ifaddrs* adapter);
 #endif
+	bool GetAdapterMAC(u8 (&mac)[6], const char* adapter_name);
```

Comment:
Other functions in this file take either `PIP_ADAPTER_ADDRESSES` or `ifaddrs*`, would it be possible to do the same here?
Although looking at `PCAPAdapter::PCAPAdapter()`, this could end up potentially duplicating some code.
It would however allow us to remove some code in `SocketAdapter::SocketAdapter()`

Reply:
Both the linux and BSD implementations don't use an `ifaddrs` here, and `PCAPAdapter::PCAPAdapter()` doesn't have one when it wants a MAC address.

Reply:
`GetGateways()` and `GetDNS()` also doesn't need `ifaddrs` over a string
It's more of a hold over from windows code, where `PIP_ADAPTER_ADDRESSES`, while containing most of the information needed, was used to identify a found/matched adapter. when porting to Linux `ifaddrs` was used as a drop in replacement.

It is, however, probably too much effort at this point to have `GetAdapterMAC()` conform to this, given the issue in `PCAPAdapter::PCAPAdapter()`

Reply:
At this point, I think the easiest way will be to have a hacky solution where a few portions in each function are overridden if the code is running on macOS. But that leads to messy, unmaintainable code.

Do you have a strategy in mind to tackle this? I am new to this codebase and I'm still learning my way around.

Reply:
It's probably more nit picking at this point I think

I do want to give the pcap code another look separately, as I've gotten a report of a game behaving differently between this and my old plugin, that may, however, just be unrelated emulation change affecting things.

Reply:
> It's probably more nit picking at this point I think
> 
> I do want to give the pcap code another look separately, as I've gotten a report of a game behaving differently between this and my old plugin, that may, however, just be unrelated emulation change affecting things.

What game was it and what changed?